### PR TITLE
Implemented Ctrl+Backspace and Ctrl+Delete for TextBox. Resolves #609

### LINKED ISSUE #609: 'Ctrl' + '←' is not working
Ctrl + Backspace should removes entire word on the left of cursor.

### PR DESCRIPTION
**Pull request by notanaverageman**

### CODE REVIEW

**Comment by grokys** [comment]
Hi @yusuf-gunaydin and thanks!

However this patch doesn't seem to work. For example pressing Ctrl+Backspace still just deletes a single character as before. I assume the patch is aiming to fix #609?

**Comment by yusuf-gunaydin** [comment]
Yes it should fix #609. I have included the test cases and also double checked in demo project. Is your case covered in unit tests?

**Comment by grokys** [comment]
Oh, apologies! It does seem to work. I think somehow VS had got confused and didn't actually build your changes.

File: src/Avalonia.Controls/TextBox.cs [hunk 1]
```diff
@@ -360,6 +360,10 @@ protected override void OnKeyDown(KeyEventArgs e)
                     break;
 
                 case Key.Back:
+		            if (modifiers == InputModifiers.Control && SelectionStart == SelectionEnd)
+		            {
+			            SetSelectionForControlBackspace(modifiers);
+		            }
                     if (!DeleteSelection() && CaretIndex > 0)
```
Comment (grokys) [inline src/Avalonia.Controls/TextBox.cs]: Nit: There seem to be some tabs in here which are screwing up alignment in the GitHub diff. Could you remove all tabs? We use all spaces. Also could you add a newline between the if statement?

File: src/Avalonia.Controls/TextBox.cs [hunk 2]
```diff
@@ -368,7 +372,11 @@ protected override void OnKeyDown(KeyEventArgs e)
                     break;
 
                 case Key.Delete:
-                    if (!DeleteSelection() && caretIndex < text.Length)
+					if (modifiers == InputModifiers.Control && SelectionStart == SelectionEnd)
+					{
+			            SetSelectionForControlDelete(modifiers);
+					}
+					if (!DeleteSelection() && caretIndex < text.Length)
```
Comment (grokys) [inline src/Avalonia.Controls/TextBox.cs]: Again, newline here.

File: src/Avalonia.Controls/TextBox.cs [hunk 3]
```diff
@@ -416,7 +424,7 @@ protected override void OnKeyDown(KeyEventArgs e)
             }
         }
 
-        protected override void OnPointerPressed(PointerPressedEventArgs e)
+	    protected override void OnPointerPressed(PointerPressedEventArgs e)
```
Comment (grokys) [inline src/Avalonia.Controls/TextBox.cs]: Tab

**Comment by grokys** [comment]
Added a few nits around whitespace, but other than that looks great - unit tests too! That's something sorely missing for TextBox as it was one of the first controls implemented :(

BTW, if you install this extension https://visualstudiogallery.msdn.microsoft.com/c8bccfe2-650c-4b42-bc5c-845e21f96328 VS should automatically pick up the whitespace settings for Avalonia from the [`.editorconfig`](https://github.com/AvaloniaUI/Avalonia/blob/master/.editorconfig) file.

File: src/Avalonia.Controls/TextBox.cs (hunk 3 above)
Comment (yusuf-gunaydin) [inline src/Avalonia.Controls/TextBox.cs]: Sure, I have checked before commiting it seemed all space. Something must have gone wrong. Fixing now.

**Comment by grokys** [comment]
Perfect! Thanks a lot for this! Will merge once CI completes.

**Pull request merged**